scripts/policy/frameworks/management/controller/main.zeek

```diff
@@ -186,6 +186,24 @@ global g_configs: table[ConfigState] of Management::Configuration
 
 function config_deploy_to_agents(config: Management::Configuration, req: Management::Request::Request)
 	{
+	# Make any final changes to the configuration we send off.
+
+	# If needed, fill in agent IP address info as learned from their peerings.
+	# XXX this will need revisiting when we support host names.
+	local instances: set[Management::Instance];
+
+	for ( inst in config$instances )
+		{
+		if ( inst$name in g_instances_known
+		    && inst$host == 0.0.0.0
+		    && g_instances_known[inst$name]$host != 0.0.0.0 )
+			inst$host = g_instances_known[inst$name]$host;
+
+		add instances[inst];
+		}
+
+	config$instances = instances;
```

Comment:
Just out of curiosity: why can't we modify the values in `config$instances` directly?

Reply:
It just doesn't work that way for sets due to how we store the indexes in the underlying dict. By the time we get an `inst`, it's not the one in the table itself.

See here: https://try.zeek.org/#/tryzeek/saved/f34d8aee749142ff8d1e11237ab22a4e

If this were a table, we could modify its values (as opposed to the keys) directly.

Reply:
Ah, right, got it.